### PR TITLE
fix issue 20915 - exclude version and debug identifier from `__traits(allMembers)` results

### PR DESCRIPTION
**Pull request by ghost**
This is the right fix this time, after clarification from @WalterBright on the semantics.

### CODE REVIEW

**Review by Geod24 — CHANGES_REQUESTED** [review]
Why not use either Condition or DVCondition ?

**Comment by NilsLankila** [comment]
Using `VersionSymbol` and `DebugSymbol` there's no need to test the identifier.

File: src/dmd/traits.d [hunk 1]
```diff
@@ -1488,6 +1488,11 @@ Expression semanticTraits(TraitsExp e, Scope* sc)
                 }
             }
 
+            // https://issues.dlang.org/show_bug.cgi?id=20915
+            // skip version and debug identifiers
+            if (sm.isVersionSymbol() || sm.isDebugSymbol())
+                return 0;
```
Comment (Geod24) [inline src/dmd/traits.d:1494]:
```suggestion
            if (sm.isDVCondition())
                return 0;
```

This is what I meant @NilsLankila . This way you go from 2 virtual calls (and 2 methods) to one.
Also you should made the `override` `final`.

Comment (NilsLankila) [reply src/dmd/traits.d:1494]: src/dmd/traits.d(1495): Error: no property `isDVCondition` for type `src/dmd.dsymbol.Dsymbol`

Comment (NilsLankila) [reply src/dmd/traits.d:1494]: And the two classes (`DebugSymbol` and `VersionSymbol`) are `final` so the override are.

Comment (NilsLankila) [reply src/dmd/traits.d:1494]: The symbols are not Condition... that cant work.

**Comment by WalterBright** [comment]
It took me a moment to figure out why the version and debug ids were even showing up in allMembers. Turns out it was inadvertent and they should be excluded. This PR looks good.

File: src/dmd/traits.d (hunk 1 above)
Comment (Geod24) [reply src/dmd/traits.d:1494]: Indeed, they are symbol equivalent, not the `Condition`, sorry for the confusion.

**Review by Geod24 — APPROVED** [review]

**Pull request merged**